src/lib/server/hyrise_communicator.cpp

```diff
@@ -0,0 +1,111 @@
+#include "hyrise_communicator.hpp"
+
+#include "expression/value_expression.hpp"
+#include "hyrise.hpp"
+#include "sql/sql_pipeline.hpp"
+#include "sql/sql_pipeline_builder.hpp"
+#include "sql/sql_translator.hpp"
+
+namespace opossum {
+
+ExecutionInformation HyriseCommunicator::execute_pipeline(const std::string& sql, const bool debug_note) {
+  // A simple query command invalidates unnamed statements
```

Comment:
I don’t get that.

Reply:
This is a PostgreSQL thing. The [documentation](https://www.postgresql.org/docs/11/protocol-flow.html#PROTOCOL-FLOW-EXT-QUERY) states:

> If successfully created, a named prepared-statement object lasts till the end of the current session, unless explicitly destroyed. An unnamed prepared statement lasts only until the next Parse statement specifying the unnamed statement as destination is issued. (Note that a simple Query message also destroys the unnamed statement.) Named prepared statements must be explicitly closed before they can be redefined by another Parse message, but this is not required for the unnamed statement. Named prepared statements can also be created and accessed at the SQL command level, using PREPARE and EXECUTE.

Gonna add a comment with a link to the documentation.